### PR TITLE
Remove legacy plugin tests

### PR DESCRIPTION
**Pull request by quinnj**
For context, see discussion in https://github.com/JuliaLang/julia/pull/53891

### CODE REVIEW

**Comment by mkitti** [comment]
Not all of those ciphers are legacy ones...

File: test/runtests.jl [hunk 1]
```diff
@@ -458,83 +458,6 @@ end
     @test _x509_certificate.issuer_name == x509_certificate.issuer_name
 end
 
-# https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html
-@testset "Encrypt" begin
-    if OpenSSL.version_number() ≥ v"3"
-        OpenSSL.load_legacy_provider()
```
Comment (fxcoudert) [inline test/runtests.jl:464]: The `load_legacy_provider` function and associated code would go away, I suppose.

File: test/runtests.jl [hunk 2]
```diff
@@ -458,83 +458,6 @@ end
...
-    end
-    evp_ciphers = [
-        EvpEncNull(),
-        EvpBlowFishCBC(), # legacy
-        EvpBlowFishECB(), # legacy
-        #EvpBlowFishCFB(), // not supported
-        EvpBlowFishOFB(), # legacy
-        EvpAES128CBC(),
```
Comment (fxcoudert) [inline test/runtests.jl:472]: `EvpAES128CBC`, `EvpAES128ECB` and `EvpAES128OFB` aren't marked as legacy and should remain.

File: test/runtests.jl [hunk 3]
```diff
@@ -458,83 +458,6 @@ end
...
-        cipher(dec_evp_cipher_ctx, enc_data, dec_data)
-        out_data = take!(dec_data)
-        out_string = String(out_data)
-
-        @test in_string == out_string
-    end
-end
-
-@testset "EncryptCustomKey" begin
-    # EvpBlowFishECB is legacy, consider using EvpAES128ECB instead
```
Comment (fxcoudert) [inline test/runtests.jl:507]: Instead of being removed, this test could be moved to a non-legacy cipher as suggested in the comment.

**Comment by giordano** [comment]
@quinnj gentle bump: could you please have a look at the comments above?  :slightly_smiling_face:

**Pull request merged**